alert-common/src/main/java/com/synopsys/integration/alert/common/action/api/AbstractConfigResourceActions.java

```diff
@@ -0,0 +1,184 @@
+
+    protected abstract ActionResponse<List<FieldModel>> readAllAfterChecks();
+
+    protected abstract ActionResponse<List<FieldModel>> readAllByContextAndDescriptorAfterChecks(String context, String descriptorName);
+
+    protected abstract Optional<FieldModel> findFieldModel(Long id);
+
+    protected abstract ValidationActionResponse testAfterChecks(FieldModel resource);
+
+    protected abstract ActionResponse<FieldModel> updateAfterChecks(Long id, FieldModel resource);
+
+    protected abstract ValidationActionResponse validateAfterChecks(FieldModel resource);
+
+    public ActionResponse<List<FieldModel>> getAllByContextAndDescriptor(String context, String descriptorName) {
+        if (!authorizationManager.hasReadPermission(context, descriptorName)) {
+            return new ActionResponse<>(HttpStatus.FORBIDDEN, AbstractResourceActions.FORBIDDEN_MESSAGE);
+        }
+        return readAllByContextAndDescriptorAfterChecks(context, descriptorName);
+    }
+
+    @Override
+    public ActionResponse<FieldModel> create(FieldModel resource) {
+        if (!authorizationManager.hasCreatePermission(resource.getContext(), resource.getDescriptorName())) {
+            return new ActionResponse<>(HttpStatus.FORBIDDEN, AbstractResourceActions.FORBIDDEN_MESSAGE);
+        }
+        ValidationActionResponse validationResponse = validateAfterChecks(resource);
+        if (validationResponse.isError()) {
+            return new ActionResponse<>(validationResponse.getHttpStatus(), validationResponse.getMessage().orElse(null));
+        }
+        return createAfterChecks(resource);
+    }
+
+    @Override
+    public ActionResponse<List<FieldModel>> getAll() {
+        try {
+            Set<String> descriptorNames = descriptorAccessor.getRegisteredDescriptors()
+                                              .stream()
+                                              .map(RegisteredDescriptorModel::getName)
+                                              .collect(Collectors.toSet());
+            if (!authorizationManager.anyReadPermission(List.of(ConfigContextEnum.DISTRIBUTION.name(), ConfigContextEnum.GLOBAL.name()), descriptorNames)) {
+                return new ActionResponse<>(HttpStatus.FORBIDDEN, AbstractResourceActions.FORBIDDEN_MESSAGE);
+            }
+            return readAllAfterChecks();
+        } catch (AlertException ex) {
+            return new ActionResponse<>(HttpStatus.INTERNAL_SERVER_ERROR, String.format("Error reading configurations: %s", ex.getMessage()));
+        }
+    }
+
+    @Override
+    public ActionResponse<FieldModel> getOne(Long id) {
+        Optional<FieldModel> fieldModel = findFieldModel(id);
+        if (fieldModel.isPresent()) {
+            FieldModel model = fieldModel.get();
+            if (!authorizationManager.hasReadPermission(model.getContext(), model.getDescriptorName())) {
+                return new ActionResponse<>(HttpStatus.FORBIDDEN, AbstractResourceActions.FORBIDDEN_MESSAGE);
```

Comment:
This statement is written a lot. Should we have a private method or static method in ActionResponse to ensure it's always the same message?